lib/cisco_node_utils/interface_channel_group.rb

```diff
@@ -62,60 +62,45 @@ def set_args_keys(hash={}) # rubocop:disable Style/AccessorMethodName
     ########################################################
 
     def channel_group_mode
-      mode = config_get('interface_channel_group', 'channel_group_mode',
-                        @get_args)
-      mode = mode ? mode : 'on' if channel_group
+      match = config_get('interface_channel_group', 'channel_group', @get_args)
+      return match unless match
+      mode = match[1].nil? ? default_channel_group_mode : match[1]
       mode
     end
 
-    def channel_group_mode=(mode)
-      group = channel_group
-
-      set_args_keys(state: 'no', group: group, force: '')
-      config_set('interface_channel_group', 'channel_group', @set_args)
+    def channel_group_mode_set(group, mode=false)
+      cgroup = channel_group
+      set_args_keys(state: 'no', group: cgroup, force: '', mode: '', val: '')
+      config_set('interface_channel_group', 'channel_group', @set_args) if
+        cgroup
+      return unless group
+      mode = false if mode && mode.to_str == 'on'
       if mode
-        Cisco::Feature.lacp_enable if mode == 'active' || mode == 'passive'
-        set_args_keys(state: '', group: group, force: 'force', mode: mode)
-        config_set('interface_channel_group', 'channel_group_mode', @set_args)
+        Cisco::Feature.lacp_enable
+        set_args_keys(state: '', group: group, force: 'force',
+                      mode: 'mode', val: mode)
+        config_set('interface_channel_group', 'channel_group', @set_args)
       else
-        set_args_keys(state: '', group: group, force: 'force')
+        set_args_keys(state: '', group: group, force: 'force',
+                      mode: '', val: '')
         config_set('interface_channel_group', 'channel_group', @set_args)
       end
     end
 
     def channel_group
-      config_get('interface_channel_group', 'channel_group', @get_args)
-    end
-
-    def channel_group=(group)
```

Comment:
not required. The whole point of this is that we just need one method which takes both channel group and mode. If mode is not specified, it will be false by default which the method already defines. 
see this:
https://github.com/cisco/cisco-network-node-utils/blob/channel_group/tests/test_interface_channel_group.rb#L94